augur/filter.py

```diff
@@ -40,18 +42,22 @@ def register_arguments(parser):
     input_group.add_argument('--metadata', required=True, metavar="FILE", help="sequence metadata, as CSV or TSV")
     input_group.add_argument('--sequences', '-s', help="sequences in FASTA or VCF format")
     input_group.add_argument('--sequence-index', help="sequence composition report generated by augur index. If not provided, an index will be created on the fly.")
-    input_group.add_argument('--metadata-chunk-size', type=int, default=100000, help="maximum number of metadata records to read into memory at a time. Increasing this number can speed up filtering at the cost of more memory used.")
+    input_group.add_argument('--metadata-chunk-size', type=int, default=100000, help="(pandas engine only) maximum number of metadata records to read into memory at a time. Increasing this number can speed up filtering at the cost of more memory used.")
     input_group.add_argument('--metadata-id-columns', default=["strain", "name"], nargs="+", help="names of valid metadata columns containing identifier information like 'strain' or 'name'")
+    input_group.add_argument('--engine', default="pandas", choices=["pandas", "sqlite"], help="Internal engine, determines syntax used for --query")
 
```

Comment:
Running some workflows using the current implementation, will add results to this thread.

Reply:
ncov-tutorial seems to run fine.

Reply:
Running ncov's rebuild-gisaid.yml:

1. Created https://github.com/nextstrain/docker-base/commit/ad3f5fbb375137750109aa079d9caa1bb7bc9c97
2. Created https://github.com/nextstrain/ncov/commit/e3490aa9509c244e419a2df490041505467015e5
3. [Ran rebuild-gisaid.yml with](https://github.com/nextstrain/ncov/actions/runs/2821555519):
    - branch: `victorlin/test-augur-filter-sqlite-engine`
    - trial build name: `victorlin-test-sqlite`
    - image: `nextstrain/base:branch-victorlin-test-augur-filter-sqlite-engine`